### PR TITLE
Improve test cases

### PR DESCRIPTION
**Pull request by nor3th**
Hey

As you mentioned in #111 and #132 you want to improve the test cases, What do you think about using the `pytest-cases` library, which I have some good experience with.

The benefit of this approach is also test for false positives (parsing results which are not intended). Like for example the `case_ipv6_addr_simple` case which also returns the ssdeep values:

- 0000:0000:ff00
- 2001:0db8:0000

Should I continue with this approach or do you want to stick to the parametrize approach from `test_asns.py`?

If you want to test the code, you can simply do that with `python3 -m pytest -s tests/test_find_iocs.py -vv`

### CODE REVIEW

**Comment by nor3th** [comment]
The 3 failing test cases are intended to fail, since I think the parsing doesn't work as intended.

File: tests/test_find_iocs.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+
+@pytest.mark.parametrize("text, results, args", ALL_TESTS)
+def test_find_iocs(text: str, results: Dict, args: Dict):
+    # Parse input
+    iocs = find_iocs(text, **args)
+
+    # Get all keys
+    ioc_finder_keys = _get_nonempty_keys(iocs)
+    # results_keys = _get_nonempty_keys(results)
```
Comment (fhightower) [inline tests/test_find_iocs.py:14]: Can we remove this line?

File: tests/test_find_iocs.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    results_keys = list(results.keys())
+
+    # Test output with expected results
+    assert sorted(ioc_finder_keys) == sorted(
+        results_keys
+    ), f"Key values don't match '{ioc_finder_keys} vs '{results_keys}'"
```
Comment (fhightower) [inline tests/test_find_iocs.py:20]: Minor, add missing quotation:

```suggestion
    ), f"Key values don't match '{ioc_finder_keys}' vs '{results_keys}'"
```

File: tests/test_find_iocs.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
+from typing import Dict, List
```
Comment (fhightower) [inline tests/test_find_iocs.py:1]: Love the use of type hints throughout this file - Thanks!

**Review by fhightower — COMMENTED** [review]
Thanks for the PR, this looks great! I left a few comments/suggestions below. Once those are addressed, this is good to go!

**Comment by fhightower** [comment]
Once merged, this fixes #132 and we'll ticket future work of porting all/most test cases into this paradigm.

**Comment by fhightower** [comment]
Thanks again for the PR 🎉 

All of the failures are tests that need updated, so I'm going to merge and we'll iterate on these over time.

**Pull request merged**